pss/pss.go

```diff
@@ -235,34 +249,41 @@ func (p *Pss) Stop() error {
 	return nil
 }
 
-var pssSpec = &protocols.Spec{
-	Name:       pssProtocolName,
-	Version:    pssVersion,
-	MaxMsgSize: defaultMaxMsgSize,
-	Messages: []interface{}{
-		PssMsg{},
-	},
-}
-
 func (p *Pss) Protocols() []p2p.Protocol {
 	return []p2p.Protocol{
 		{
-			Name:    pssSpec.Name,
-			Version: pssSpec.Version,
-			Length:  pssSpec.Length(),
+			Name:    spec.Name,
+			Version: spec.Version,
+			Length:  spec.Length(),
 			Run:     p.Run,
 		},
 	}
 }
 
 func (p *Pss) Run(peer *p2p.Peer, rw p2p.MsgReadWriter) error {
-	pp := protocols.NewPeer(peer, rw, pssSpec)
-	p.fwdPoolMu.Lock()
-	p.fwdPool[peer.Info().ID] = pp
-	p.fwdPoolMu.Unlock()
+	pp := protocols.NewPeer(peer, rw, spec)
+	p.addPeer(pp)
+	defer p.removePeer(pp)
 	return pp.Run(p.handlePssMsg)
 }
 
+func (p *Pss) addPeer(peer *protocols.Peer) {
+	p.peersMu.Lock()
+	defer p.peersMu.Unlock()
+	p.peers[peer.Peer.Info().ID] = peer
+}
+
+func (p *Pss) removePeer(peer *protocols.Peer) {
+	p.peersMu.Lock()
+	defer p.peersMu.Unlock()
+	if _, found := p.peers[peer.Peer.Info().ID]; found {
```

Comment:
not needed, just delete :)

Reply:
Why do you think it is not needed?

Reply:
I agree. It's really just deleting from a map. I can't recall any client code in which we care whether it's there or not.